### PR TITLE
Dataset histogram

### PR DESCRIPTION
**Pull request by nick-Ag**
image
Histograms have been moved from the style options menu to the dataset options menu. When the user toggles on a histogram, a new tab will be created and a histogram will be generated by plotly in that tab. The x-axis is the range of values of the data channel and the y-axis is the count of values in the range. No custom bin sizes yet, for time purposes.

### CODE REVIEW

File: frontend/src/app/side-menu/side-menu.component.ts [hunk 1]
```diff
@@ -148,7 +128,7 @@ export class SideMenuComponent {
   uploadFiles() {
     const fileUpload = this.fileUpload.nativeElement;
     if (this.checkBox.checked) {
-      this.dashboard.newTab();
+      this.dashboard.newTab('New Tab');
```
Comment (rglasser) [inline frontend/src/app/side-menu/side-menu.component.ts:131]: Is this user editable at all, or will it always just say "New Tab"?

Comment (nick-Ag) [reply frontend/src/app/side-menu/side-menu.component.ts:131]: I was going to do a general UI update fix to address that plus a few other things. I don't think I have time to make them user editable so they'll just say something like 'TMD2725 Ambient Light'

File: frontend/src/app/dataset/dataset.component.ts [hunk 2]
```diff
@@ -217,11 +240,14 @@ export class DatasetComponent {
...
-      const tab = channel === 'stdev' ? this.tabNumbers[1] : this.tabNumbers[0];
+      const tab =
+        channel === 'stdev'
+          ? this.tabNumbers.get('stdev')
+          : this.tabNumbers.get('plot');
       const id = toggleStats
         ? channel + this.currentOptions
         : String(this.currentOptions);
-
+
```
Comment (rglasser) [inline frontend/src/app/dataset/dataset.component.ts:250]: nit: remove this extra space

File: frontend/src/app/main-dashboard/main-dashboard.component.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ import {Component, ViewChildren, QueryList, AfterViewInit} from '@angular/core';
 
 // Project Imports.
 import {PlotComponent} from '../plot/plot.component';
+import { MatTab } from '@angular/material/tabs';
```
Comment (agarwal-v) [inline frontend/src/app/main-dashboard/main-dashboard.component.ts:20]: inconsistent spacing between lines 19 & 20

Comment (agarwal-v) [reply frontend/src/app/main-dashboard/main-dashboard.component.ts:20]: More specifically around {}

File: frontend/src/app/style-dialog/style-dialog.component.html [hunk 4]
```diff
@@ -15,7 +15,8 @@
 <h1>Style Options</h1>
 <h3>Trace: {{traceName}}</h3>
 
-<!-- Toggle: Scatter , Line, Histogram -->
+<!-- Toggle: Scatter , Line-->
```
Comment (agarwal-v) [inline frontend/src/app/style-dialog/style-dialog.component.html:18]: space after "Line"

**Review by agarwal-v — COMMENTED** [review]
minor formatting changes

**Pull request merged**